assets/js/invidious_iframe_api.js

```diff
@@ -0,0 +1,724 @@
+class invidious_embed {
+    static widgetid = 0;
+    static eventname_table = { onPlaybackRateChange: 'ratechange', onStateChange: 'statechange', onError: 'error', onReady: 'ready' };
+    static available_event_name = ['ready', 'ended', 'error', 'ratechange', 'volumechange', 'waiting', 'timeupdate', 'loadedmetadata', 'play', 'seeking', 'seeked', 'playerresize', 'pause'];
+    static api_promise = false;
+    static invidious_instance = '';
+    static api_instance_list = [];
+    static instance_status_list = {};
+    static videodata_cahce = {};
+
+    addEventListener(eventname, func) {
+        if (typeof func === 'function') {
+            if (eventname in invidious_embed.eventname_table) {
+                this.eventobject[invidious_embed.eventname_table[eventname]].push(func);
+            } else if (invidious_embed.available_event_name.includes(eventname)) {
+                this.eventobject[eventname].push(func);
+            } else {
+                console.warn('addEventListener cannot find such eventname : ' + eventname);
+            }
+        } else {
+            console.warn("addEventListner secound args must be function");
+        }
+    }
+
+    removeEventListener(eventname, func) {
+        if (typeof func === 'function') {
+            let internal_eventname;
+            if (eventname in invidious_embed.eventname_table) {
+                internal_eventname = invidious_embed.eventname_table[eventname];
+            } else if (invidious_embed.available_event_name.includes(eventname)) {
+                internal_eventname = eventname;
+            } else {
+                console.warn('removeEventListner cannot find such eventname : ' + eventname);
+                return;
+            }
+            this.eventobject[internal_eventname] = this.eventobject[internal_eventname].filter(x => {
+                const allowFunctionDetected = x.toString()[0] === '(';
+                if (allowFunctionDetected) {
+                    x.toString() !== func.toString();
+                } else {
+                    x !== func;
+                }
+            });
+        } else {
+            console.warn("removeEventListener secound args must be function");
+        }
+    }
+
+    async instance_access_check(instance_origin) {
+        let return_status;
+        const status_cahce_exist = instance_origin in invidious_embed.instance_status_list;
+        if (!status_cahce_exist) {
+            try {
+                const instance_stats = await fetch(instance_origin + '/api/v1/stats');
+                if (instance_stats.ok) {
+                    const instance_stats_json = await instance_stats.json();
+                    if (instance_stats_json.software.name === 'invidious') {
+                        return_status = true;
+                    } else {
+                        return_status = false;
+                    }
```

Comment:
The strict equality (===) operator always returns a boolean, so you can use that directly in your variable:
```suggestion
                    return_status = (instance_stats_json.software.name === 'invidious');
```